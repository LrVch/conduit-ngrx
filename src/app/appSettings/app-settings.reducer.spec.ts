import * as fromAppSettings from './app-settings.reducer';
import * as AppSettingsActions from './app-settings.actions';
import { AppSettingsState } from '@app/core/models/app-settings.model';

describe('Article Reducer', () => {
  const initialAppSettingsState: AppSettingsState = {
    ...fromAppSettings.initialState,
    language: 'en'
  };

  describe('State changes', () => {
    it('should have an initial state', () => {
      const state = fromAppSettings.appSettingsReducer(
        initialAppSettingsState,
        {
          type: '@@init'
        } as any
      );

      expect(state).toBe(initialAppSettingsState);
    });

    it('should change language', () => {
      const init = { type: '@@init' } as any;
      const action = new AppSettingsActions.AppSettingsChangeLanguage({
        language: 'ru'
      });

      const state = [init, action].reduce(
        fromAppSettings.appSettingsReducer,
        initialAppSettingsState
      );

      expect(state).toMatchSnapshot();
    });

    it('should change hour', () => {
      const init = { type: '@@init' } as any;
      const action = new AppSettingsActions.AppSettingsChangeHour({ hour: 23 });

      const state = [init, action].reduce(
        fromAppSettings.appSettingsReducer,
        initialAppSettingsState
      );

      expect(state).toMatchSnapshot();
    });

    it('should change theme', () => {
      const init = { type: '@@init' } as any;
      const action = new AppSettingsActions.AppSettingsChangeTheme({
        theme: 'BLACK-THEME'
      });

      const state = [init, action].reduce(
        fromAppSettings.appSettingsReducer,
        initialAppSettingsState
      );

      expect(state).toMatchSnapshot();
    });

    it('should change autoNightMode', () => {
      const init = { type: '@@init' } as any;
      const action = new AppSettingsActions.AppSettingsChangeAutoNightMode({
        autoNightMode: true
      });

      const state = [init, action].reduce(
        fromAppSettings.appSettingsReducer,
        initialAppSettingsState
      );

      expect(state).toMatchSnapshot();
    });

    it('should change stickyHeader', () => {
      const init = { type: '@@init' } as any;
      const action = new AppSettingsActions.AppSettingsChangeStickyHeader({
        stickyHeader: false
      });

      const state = [init, action].reduce(
        fromAppSettings.appSettingsReducer,
        initialAppSettingsState
      );

      expect(state).toMatchSnapshot();
    });

    it('should change asideOpenMode', () => {
      const init = { type: '@@init' } as any;
      const action = new AppSettingsActions.AppSettingsChangeAsideOpenMode({
        asideOpenMode: 'over'
      });

      const state = [init, action].reduce(
        fromAppSettings.appSettingsReducer,
        initialAppSettingsState
      );

      expect(state).toMatchSnapshot();
    });

    it('should change nightModeFrom', () => {
      const init = { type: '@@init' } as any;
      const action = new AppSettingsActions.AppSettingsChangeNightModeFrom({
        nightModefrom: 17
      });

      const state = [init, action].reduce(
        fromAppSettings.appSettingsReducer,
        initialAppSettingsState
      );

      expect(state).toMatchSnapshot();
    });

    it('should change nightModeTo', () => {
      const init = { type: '@@init' } as any;
      const action = new AppSettingsActions.AppSettingsChangeNightModeTo({
        nightModeto: 17
      });

      const state = [init, action].reduce(
        fromAppSettings.appSettingsReducer,
        initialAppSettingsState
      );

      expect(state).toMatchSnapshot();
    });

    it('should change effectiveTheme', () => {
      const init = { type: '@@init' } as any;
      const action = new AppSettingsActions.AppSettingsUpdateEffectiveTheme({
        effectiveTheme: 'BLACK-THEME'
      });

      const state = [init, action].reduce(
        fromAppSettings.appSettingsReducer,
        initialAppSettingsState
      );

      expect(state).toMatchSnapshot();
    });

    it('should change routeAnimationEnabled', () => {
      const init = { type: '@@init' } as any;
      const action = new AppSettingsActions.AppSettingsChangeRouteAnimationEnabled(
        { routeAnimationEnabled: false }
      );

      const state = [init, action].reduce(
        fromAppSettings.appSettingsReducer,
        initialAppSettingsState
      );

      expect(state).toMatchSnapshot();
    });

    it('should change routeAnimationsChangeType', () => {
      const init = { type: '@@init' } as any;
      const action = new AppSettingsActions.AppSettingsChangeRouteAnimationType(
        { routeAnimationsChangeType: 'FADE' }
      );

      const state = [init, action].reduce(
        fromAppSettings.appSettingsReducer,
        initialAppSettingsState
      );

      expect(state).toMatchSnapshot();
    });
  });
});
