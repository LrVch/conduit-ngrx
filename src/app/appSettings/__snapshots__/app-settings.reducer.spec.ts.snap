// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Article Reducer State changes should change asideOpenMode 1`] = `
Object {
  "asideOpenMode": "over",
  "asideOpenModes": Array [
    "push",
    "over",
  ],
  "autoNightMode": false,
  "defaultLanguage": "en",
  "effectiveTheme": null,
  "hour": 0,
  "language": "en",
  "languages": Array [
    "en",
    "ru",
  ],
  "nightModefrom": 21,
  "nightModeto": 7,
  "nightTheme": "BLACK-THEME",
  "routeAnimationEnabled": true,
  "routeAnimationsChangeType": "SLIDE",
  "routeAnimationsChangeTypes": Array [
    "SLIDE",
    "FADE",
  ],
  "stickyHeader": true,
  "theme": "DEFAULT-THEME",
  "themes": Array [
    "DEFAULT-THEME",
    "BLUE-THEME",
    "BLACK-THEME",
  ],
}
`;

exports[`Article Reducer State changes should change autoNightMode 1`] = `
Object {
  "asideOpenMode": "push",
  "asideOpenModes": Array [
    "push",
    "over",
  ],
  "autoNightMode": true,
  "defaultLanguage": "en",
  "effectiveTheme": null,
  "hour": 0,
  "language": "en",
  "languages": Array [
    "en",
    "ru",
  ],
  "nightModefrom": 21,
  "nightModeto": 7,
  "nightTheme": "BLACK-THEME",
  "routeAnimationEnabled": true,
  "routeAnimationsChangeType": "SLIDE",
  "routeAnimationsChangeTypes": Array [
    "SLIDE",
    "FADE",
  ],
  "stickyHeader": true,
  "theme": "DEFAULT-THEME",
  "themes": Array [
    "DEFAULT-THEME",
    "BLUE-THEME",
    "BLACK-THEME",
  ],
}
`;

exports[`Article Reducer State changes should change effectiveTheme 1`] = `
Object {
  "asideOpenMode": "push",
  "asideOpenModes": Array [
    "push",
    "over",
  ],
  "autoNightMode": false,
  "defaultLanguage": "en",
  "effectiveTheme": "BLACK-THEME",
  "hour": 0,
  "language": "en",
  "languages": Array [
    "en",
    "ru",
  ],
  "nightModefrom": 21,
  "nightModeto": 7,
  "nightTheme": "BLACK-THEME",
  "routeAnimationEnabled": true,
  "routeAnimationsChangeType": "SLIDE",
  "routeAnimationsChangeTypes": Array [
    "SLIDE",
    "FADE",
  ],
  "stickyHeader": true,
  "theme": "DEFAULT-THEME",
  "themes": Array [
    "DEFAULT-THEME",
    "BLUE-THEME",
    "BLACK-THEME",
  ],
}
`;

exports[`Article Reducer State changes should change hour 1`] = `
Object {
  "asideOpenMode": "push",
  "asideOpenModes": Array [
    "push",
    "over",
  ],
  "autoNightMode": false,
  "defaultLanguage": "en",
  "effectiveTheme": null,
  "hour": 23,
  "language": "en",
  "languages": Array [
    "en",
    "ru",
  ],
  "nightModefrom": 21,
  "nightModeto": 7,
  "nightTheme": "BLACK-THEME",
  "routeAnimationEnabled": true,
  "routeAnimationsChangeType": "SLIDE",
  "routeAnimationsChangeTypes": Array [
    "SLIDE",
    "FADE",
  ],
  "stickyHeader": true,
  "theme": "DEFAULT-THEME",
  "themes": Array [
    "DEFAULT-THEME",
    "BLUE-THEME",
    "BLACK-THEME",
  ],
}
`;

exports[`Article Reducer State changes should change language 1`] = `
Object {
  "asideOpenMode": "push",
  "asideOpenModes": Array [
    "push",
    "over",
  ],
  "autoNightMode": false,
  "defaultLanguage": "en",
  "effectiveTheme": null,
  "hour": 0,
  "language": "ru",
  "languages": Array [
    "en",
    "ru",
  ],
  "nightModefrom": 21,
  "nightModeto": 7,
  "nightTheme": "BLACK-THEME",
  "routeAnimationEnabled": true,
  "routeAnimationsChangeType": "SLIDE",
  "routeAnimationsChangeTypes": Array [
    "SLIDE",
    "FADE",
  ],
  "stickyHeader": true,
  "theme": "DEFAULT-THEME",
  "themes": Array [
    "DEFAULT-THEME",
    "BLUE-THEME",
    "BLACK-THEME",
  ],
}
`;

exports[`Article Reducer State changes should change nightModeFrom 1`] = `
Object {
  "asideOpenMode": "push",
  "asideOpenModes": Array [
    "push",
    "over",
  ],
  "autoNightMode": false,
  "defaultLanguage": "en",
  "effectiveTheme": null,
  "hour": 0,
  "language": "en",
  "languages": Array [
    "en",
    "ru",
  ],
  "nightModefrom": 17,
  "nightModeto": 7,
  "nightTheme": "BLACK-THEME",
  "routeAnimationEnabled": true,
  "routeAnimationsChangeType": "SLIDE",
  "routeAnimationsChangeTypes": Array [
    "SLIDE",
    "FADE",
  ],
  "stickyHeader": true,
  "theme": "DEFAULT-THEME",
  "themes": Array [
    "DEFAULT-THEME",
    "BLUE-THEME",
    "BLACK-THEME",
  ],
}
`;

exports[`Article Reducer State changes should change nightModeTo 1`] = `
Object {
  "asideOpenMode": "push",
  "asideOpenModes": Array [
    "push",
    "over",
  ],
  "autoNightMode": false,
  "defaultLanguage": "en",
  "effectiveTheme": null,
  "hour": 0,
  "language": "en",
  "languages": Array [
    "en",
    "ru",
  ],
  "nightModefrom": 21,
  "nightModeto": 17,
  "nightTheme": "BLACK-THEME",
  "routeAnimationEnabled": true,
  "routeAnimationsChangeType": "SLIDE",
  "routeAnimationsChangeTypes": Array [
    "SLIDE",
    "FADE",
  ],
  "stickyHeader": true,
  "theme": "DEFAULT-THEME",
  "themes": Array [
    "DEFAULT-THEME",
    "BLUE-THEME",
    "BLACK-THEME",
  ],
}
`;

exports[`Article Reducer State changes should change routeAnimationEnabled 1`] = `
Object {
  "asideOpenMode": "push",
  "asideOpenModes": Array [
    "push",
    "over",
  ],
  "autoNightMode": false,
  "defaultLanguage": "en",
  "effectiveTheme": null,
  "hour": 0,
  "language": "en",
  "languages": Array [
    "en",
    "ru",
  ],
  "nightModefrom": 21,
  "nightModeto": 7,
  "nightTheme": "BLACK-THEME",
  "routeAnimationEnabled": false,
  "routeAnimationsChangeType": "SLIDE",
  "routeAnimationsChangeTypes": Array [
    "SLIDE",
    "FADE",
  ],
  "stickyHeader": true,
  "theme": "DEFAULT-THEME",
  "themes": Array [
    "DEFAULT-THEME",
    "BLUE-THEME",
    "BLACK-THEME",
  ],
}
`;

exports[`Article Reducer State changes should change routeAnimationsChangeType 1`] = `
Object {
  "asideOpenMode": "push",
  "asideOpenModes": Array [
    "push",
    "over",
  ],
  "autoNightMode": false,
  "defaultLanguage": "en",
  "effectiveTheme": null,
  "hour": 0,
  "language": "en",
  "languages": Array [
    "en",
    "ru",
  ],
  "nightModefrom": 21,
  "nightModeto": 7,
  "nightTheme": "BLACK-THEME",
  "routeAnimationEnabled": true,
  "routeAnimationsChangeType": "FADE",
  "routeAnimationsChangeTypes": Array [
    "SLIDE",
    "FADE",
  ],
  "stickyHeader": true,
  "theme": "DEFAULT-THEME",
  "themes": Array [
    "DEFAULT-THEME",
    "BLUE-THEME",
    "BLACK-THEME",
  ],
}
`;

exports[`Article Reducer State changes should change stickyHeader 1`] = `
Object {
  "asideOpenMode": "push",
  "asideOpenModes": Array [
    "push",
    "over",
  ],
  "autoNightMode": false,
  "defaultLanguage": "en",
  "effectiveTheme": null,
  "hour": 0,
  "language": "en",
  "languages": Array [
    "en",
    "ru",
  ],
  "nightModefrom": 21,
  "nightModeto": 7,
  "nightTheme": "BLACK-THEME",
  "routeAnimationEnabled": true,
  "routeAnimationsChangeType": "SLIDE",
  "routeAnimationsChangeTypes": Array [
    "SLIDE",
    "FADE",
  ],
  "stickyHeader": false,
  "theme": "DEFAULT-THEME",
  "themes": Array [
    "DEFAULT-THEME",
    "BLUE-THEME",
    "BLACK-THEME",
  ],
}
`;

exports[`Article Reducer State changes should change theme 1`] = `
Object {
  "asideOpenMode": "push",
  "asideOpenModes": Array [
    "push",
    "over",
  ],
  "autoNightMode": false,
  "defaultLanguage": "en",
  "effectiveTheme": null,
  "hour": 0,
  "language": "en",
  "languages": Array [
    "en",
    "ru",
  ],
  "nightModefrom": 21,
  "nightModeto": 7,
  "nightTheme": "BLACK-THEME",
  "routeAnimationEnabled": true,
  "routeAnimationsChangeType": "SLIDE",
  "routeAnimationsChangeTypes": Array [
    "SLIDE",
    "FADE",
  ],
  "stickyHeader": true,
  "theme": "BLACK-THEME",
  "themes": Array [
    "DEFAULT-THEME",
    "BLUE-THEME",
    "BLACK-THEME",
  ],
}
`;
